test/tools/spec-runner/index.js

```diff
@@ -390,18 +390,13 @@ function runTestSuiteTest(configuration, spec, context) {
     let testPromise = Promise.resolve();
     return testPromise
       .then(() => testOperations(spec, operationContext))
-      .catch(err => {
-        // If the driver throws an exception / returns an error while executing this series
-        // of operations, store the error message.
-        throw err;
-      })
-      .then(() => {
+      .then(() => validateExpectations(context.commandEvents, spec, savedSessionData))
```

Comment:
we probably want to reverse this order, since that's what is making the spec tests here fail; does reversing the order back to what it was cause other tests to fail?